indigo_social/views.py

```diff
@@ -24,21 +24,23 @@ class ContributorsView(ListView):
 
 class UserProfileView(DetailView):
     model = UserProfile
```

Comment:
I think this view's model should be the `User` object, not `UserProfile`. After all, the view encompasses the whole user (eg. first name, username, etc.).

That would mean you don't need to special-case `get_object`.

In the template, the profile is available as `user.userprofile`.

Reply:
I was lazy to change the template but I see the very good reason for doing that ( :